jobs/kola-openstack.Jenkinsfile

```diff
@@ -0,0 +1,111 @@
+        if (meta.images.openstack) {
+            openstack_image_filename = meta.images.openstack.path
+            openstack_image_sha256 = meta.images.openstack.sha256
+            openstack_image_filepath = "builds/${params.VERSION}/${basearch}/${openstack_image_filename}"
+        } else {
+            throw new Exception("No OpenStack artifacts found in metadata for ${params.VERSION}")
+        }
+
+        // Copy down the openstack image from S3, verify, uncompress
+        // In the future maybe we can use `cosa fetch-upstream` here instead.
+        // https://github.com/coreos/coreos-assembler/issues/1508
+        shwrap("""
+        export AWS_CONFIG_FILE=\${AWS_FCOS_BUILDS_BOT_CONFIG}/config
+        aws s3 cp --no-progress s3://${s3_stream_dir}/${openstack_image_filepath} ${openstack_image_filepath}
+        echo "${openstack_image_sha256} ${openstack_image_filepath}" | sha256sum --check
+        unxz ${openstack_image_filepath}
+        """)
+        // Remove '.xz` from the end of the filename and file path
+        openstack_image_filename = openstack_image_filename[0..-4]
+        openstack_image_filepath = openstack_image_filepath[0..-4]
+        // Prepend the image name with 'kola', in case it gets left behind
+        openstack_image_name = "kola-${openstack_image_filename}"
+
+    }
+
+    stage('Upload/Create Image') {
+        // Create the image in OpenStack
+        shwrap("""
+        ore openstack --config-file=\${OPENSTACK_KOLA_TESTS_CONFIG}/config \
+             create-image --file=${openstack_image_filepath} \
+             --name=${openstack_image_name}
+        """)
+    }
+    
+    // In VexxHost we'll use the network called "private" for the
+    // instance NIC, attach a floating IP from the "public" network and
+    // use the v1-standard-4 instance (ram=4GiB, disk=100GiB, CPUs=4).
+    // The clouds.yaml config should be located at ${OPENSTACK_KOLA_TESTS_CONFIG}/config.
+    //
+    // Since we don't have permanent images uploaded to VexxHost we'll
+    // skip the upgrade test.
+    try {
+        fcosKola(cosaDir: env.WORKSPACE, parallel: 5,
+                 build: params.VERSION, skipUpgrade: true,
+                 platformArgs: """-p=openstack                               \
+                    --openstack-config-file=\${OPENSTACK_KOLA_TESTS_CONFIG}/config \
+                    --openstack-flavor=v1-standard-4                         \
+                    --openstack-network=private                              \
+                    --openstack-floating-ip-network=public                   \
+                    --openstack-image=${openstack_image_name}""")
+    } finally {
+        stage('Delete Image') {
+            // Delete the image in OpenStack
+            shwrap("""
+            ore openstack --config-file=\${OPENSTACK_KOLA_TESTS_CONFIG}/config \
+                 delete-image --id=${openstack_image_name}
+            """)
+        }
```

Comment:
Would be nice if there were a more foolproof way to do this so that we're sure we'll never leak images. E.g. right now if this job gets interrupted before we delete the image, it'll never get cleaned up.

Hmm, how about this: name the image `${params.STREAM}` and delete any image of that name before creating it (using `ore openstack delete-image`). That way, we're sure that any leaks eventually gets cleaned up. The downside is that we wouldn't be able to run multiple jobs testing the same stream at different versions, though not sure how likely that is.

Reply:
added a new commit to do this.